src/tests/math/test_linspace.f90

```diff
@@ -0,0 +1,397 @@
+        call check(size(x) == n, msg="Array not allocated to appropriate size", warn=warn)
+
+        print *, "Made it through first round of tests"
+
+        ! Due to roundoff error, it is possible that the jump from x(n-1) to x(n) is slightly different than the expected interval
+        do i = 1, n-1
+
+            true_difference = x(i + 1) - x(i)
+            call check(abs(true_difference - expected_interval) < abs(expected_interval) * TOLERANCE_SP)
+
+        end do
+
+    end subroutine
+
+    subroutine test_linspace_dp
+
+        real(dp) :: start = 1.0_dp
+        real(dp) :: end = 10.0_dp
+        integer, parameter :: n = DEFAULT_LINSPACE_LENGTH
+        real(dp) :: expected_interval
+        real(dp) :: true_difference
+
+        real(dp), allocatable :: x(:)
+        integer :: i
+
+        x = linspace(start, end)
+
+        expected_interval =( end - start ) / ( n - 1 )
+
+        call check(x(1) == start, msg="Initial value of array is not equal to the passed start parameter", warn=warn)
+        call check(x(n) == end, msg="Final array value is not equal to end parameter", warn=warn)
+        call check(size(x) == n, msg="Array not allocated to default size", warn=warn)
+
+        ! Due to roundoff error, it is possible that the jump from x(n-1) to x(n) is slightly different than the expected interval
+        do i = 1, n-1
+
+            true_difference = x(i + 1) - x(i)
+            call check(abs(true_difference - expected_interval) < abs(expected_interval) * TOLERANCE_DP)
+
+        end do
+
+    end subroutine
+
+    subroutine test_linspace_neg_index
+
+        real(dp) :: start = 1.0_dp
+        real(dp) :: end = 10.0_dp
+
+        real(dp), allocatable :: x(:)
+
+        x = linspace(start, end, -15)
+
+        call check(size(x) == 0, msg="Allocated array is not empty", warn=warn)
+
+    end subroutine
+
+    subroutine test_linspace_cmplx
+
+        complex(dp) :: start = (0.0_dp, 10.0_dp)
+        complex(dp) :: end = (1.0_dp, 0.0_dp)
+        complex(dp) :: expected_interval
+        integer, parameter :: n = 10
+
+        complex(dp), allocatable :: z(:)
+
+        integer :: i
+
+        z = linspace(start, end, n)
+
+        expected_interval =( end - start ) / ( n - 1 )
+
+        call check(z(1) == start, msg="Initial value of array is not equal to the passed start parameter", warn=warn)
+        call check(z(n) == end, msg="Final array value is not equal to end parameter", warn=warn)
+        call check(size(z) == n, msg="Array not allocated to correct size", warn=warn)
+
+        ! Due to roundoff error, it is possible that the jump from x(n-1) to x(n) is slightly different than the expected interval
+        do i = 1, n-1
+
+            call check(abs( ( z(i + 1) - z(i) ) - expected_interval) < abs(expected_interval) * TOLERANCE_DP)
+
+        end do
+
+        write(unit=iunit, fmt=*) "linspace((0.0_dp, 10.0_dp), (1.0_dp, 0.0_dp), 10): "
+        write(unit=iunit,fmt=99)
```

Comment:
Here, and in all other instances (13 total), I suggest specifying the format in the `write` statement instead of the `format` statement below:

```suggestion
        write(unit=iunit,fmt='(70("="))')
```

Rationale: the format is referred to only once in each test subroutine, and IMO it's harder to read having to jump down to look for a labeled statement. 

Ideally, as this format is used several times in each test program, could make it a character constant and reuse:

```
character(70), parameter :: delimiter_fmt = repeat('=', 70)
```

But this would require a module, so may not be worth it at this time.

Reply:
Thanks for the suggestion. I will fix this shortly.

Reply:
@ejovo13 I think that this is the really latest minor issue with this PR. When fixed, don't hesitate to ping one of us for merging it.

Reply:
Thank you. I will get this fixed and ping you when it's cleaned up. I have been off the grid camping so expect a message within the next week!

Reply:
I cleaned up the format statements as suggested by @milancurcic. If all checks pass, thank you everyone for the reviews and @jvdp1 please go ahead with the merge!

Reply:
Great. I will merge it. Thank you @ejovo13